mel_jamei/week_01/thursday/homework03.js

```diff
@@ -0,0 +1,212 @@
+/*Geometry Function Lab
+
+Part 1, Rectangle
+
+Given the following a rectangle object like the one below, write the following functions:
+
+isSquare - Returns whether the rectangle is a square or not*/
+var rectangle = {
+  length: 3,
+  width: 4
+};
+
+var isSquare = function ( rect ) {
+
+    if (rect.length === rect.width ) {
+
+      console.log('this is a square');
+
+      return true;
+
+  } else {
+
+      console.log('this is not a square');
+  }
+};
+//area - Returns the area of the rectangle
+
+var area = function ( rectan ) {
+
+    var total = rectan.length * rectan.width;
+
+    return total;
+}
+
+/*perimeter - Returns the perimeter of the rectangle
+
+};*/
+
+var perimeter = function ( input ) {
+
+    //p = 2 * (length + width)
+
+    var total =  2 * (input.length + input.width);
+
+    return total;
+
+}
+
+isSquare(rectangle);
+area(rectangle);
+perimeter(input);
+
+
+//Part 2, Triangle
+
+//Given the following a triangle object like the one below, write the following functions:
+
+
+//isEquilateral - Returns whether the triangle is equilateral or not
+
+var triangle = {
+  sideA: 3,
+  sideB: 4,
+  sideC: 4
+};
+
+
+
+var isEquilateral = function (triangle) {
+
+  var a = triangle.sideA;
+  var b = triangle.sideB;
+  var c = triangle.sideC;
+
+  if (a === b && b === c && a === c) {
+
+      console.log('This is an equilateral triangle');
+
+  } else {``
+
+      console.log('This is not an equilateral triangle');
+  }
+}
+
+//isIsosceles - Returns whether the triangle is isosceles or not
+
+var isIsosceles = function (triangle) {
+
+  var a = triangle.sideA;
+  var b = triangle.sideB;
+  var c = triangle.sideC;
+
+    if (!(a===b)c){
+
+      console.log('yes');
+
+    } else {
+
+      console.log('no');
+    }
+}
+
+//area - Returns the area of the Triangle
+var area = function (triangle) { // 1/2 * h * b
+
+    var total = 1/2 * triangle.sideB * triangle.heigth;
+
+    return total;
+
+}
+
+//isObtuse - Returns whether the triangle is obtuse or not
+
+var isObtuse = function (){
+
+}
+
+
+
+/*The Cash Register
+
+Write a function called cashRegister that takes a shopping cart object. The object contains item names and prices (itemName: itemPrice). The function should return the total price of the shopping cart. Example
+
+// Input
+var cartForParty = {
+  banana: "1.25",
+  handkerchief: ".99",
+  Tshirt: "25.01",
+  apple: "0.60",
+  nalgene: "10.34",
+  proteinShake: "22.36"
+};
+
+// Output
+cashRegister(cartForParty)); // 60.55*/
+
+
+var cashRegister = function (cash){
+
+    var input = Object.values(cash);// to get just the values out of object
+
+    var dollars = 0;// to start counting from zero
+
+        for (var i = 0; i < input.length; i++) {
+          dollars + = parseFloat(input[i])// parseFlaot to convert strings to decimal numbers
```

Comment:
Be careful with that `+ =`. Leaving a space between them is a syntax error. It has to be written as `+=`, no space.